src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -614,9 +615,162 @@ protected void _addFields(Map<String, POJOPropertyBuilder> props)
+        Iterator<PotentialCreator> it = ctors.iterator();
+        while (it.hasNext()) {
+            PotentialCreator ctor = it.next();
+
+            final int paramCount = ctor.paramCount();
+            if (paramCount == 0) {
+                it.remove();
+                collector.addDefault(ctor.creator);
+                continue;
+            }
+            
+            if (ctor.creatorMode == null) {
+                continue;
+            }
+            it.remove();
+
+            Boolean propsBased = null;
+            
+            switch (ctor.creatorMode) {
+            case DELEGATING:
+                propsBased = false;
+                break;
+            case PROPERTIES:
+                propsBased = true;
+                break;
+            case DEFAULT:
+            default:
+                // First things first: if not single-arg Creator, must be Properties-based
+                // !!! Or does it? What if there's @JacksonInject etc?
+                if (paramCount != 1) {
+                    propsBased = true;
+                }
+            }
+
+            // Must be 1-arg case:
+            if (propsBased == null) {
+                // Is ambiguity/heuristics allowed?
+                if (ctorDetector.requireCtorAnnotation()) {
+                    throw new IllegalArgumentException(String.format(
+                            "Ambiguous 1-argument Creator; `ConstructorDetector` requires specifying `mode`: %s",
+                            ctor));
+                }
+    
+                // First things first: if explicit names found, is Properties-based
+                ctor.introspectParamNames(_config);
+                propsBased = ctor.hasExplicitNames()
+                        || ctorDetector.singleArgCreatorDefaultsToProperties();
+                // One more possibility: implicit name that maps to implied
+                // property based on Field/Getter/Setter
+                if (!propsBased) {
+                    String implName = ctor.implicitNameSimple(0);
+                    propsBased = (implName != null) && props.containsKey(implName);
+                }
+            }
+
+            if (propsBased) {
+                // Skipping done if we already got higher-precedence Creator
+                if (!skipPropsBased) {
+                    collector.addPropertiesBased(_config, ctor, "explicit");
+                }
+            } else {
+                collector.addDelegating(ctor);
+            }
+        }
+    }
+
+    private void _addExplicitCreatorParams(Map<String, POJOPropertyBuilder> props,
+            PotentialCreator ctor)
+    {
+        for (int i = 0, len = ctor.paramCount(); i < len; ++i) {
+            final AnnotatedParameter param = ctor.param(i);
+            final PropertyName explName = ctor.explicitName(i);
+            PropertyName implName = ctor.implicitName(i);
+            final boolean hasExplicit = (explName != null);
+
+            if (!hasExplicit) {
```

Comment:
This may be personal preference.

From here to line 766 seems like we can just end everything with...

    if  explicit do explcit handling
    else do implicit 

And not mix up two different handling

Reply:
Yeah I am sure there are ways to clean things up: at first I am trying to replicate existing logic. So once things pass will focus more on clean up.

Exciting thing is that right now I am 1 test fail away from passing all Java 8 tests -- then need to tackle record handling.

And once that happens, then start making use of Creators detected: right now all I have changed is finding properties-based Creators to link properties in `POJOPropertiesCollector`, but `BasicDeserializerFactory` then re-scans everything. But that won't be necessary for long.

... and need to figure out at what point to merge things to `master`. Probably half-way through, to keep diff manageable.

But at least I am FINALLY making progress!

Reply:
Forgot to mention that this part of code will definitely be rewritten as right now it is strictly duplicating existing handling, but once everything works as well as it used to, can and will start simplifying & cleaning up.